Dockerfile

```diff
@@ -0,0 +1,54 @@
+
+# s2i-haproxy
+FROM openshift/base-centos7
+
+ENV HAPROXY_DOWNLOAD "https://www.haproxy.org/download/1.8/src/haproxy-1.8.0.tar.gz"
+ENV HAPROXY_MD5 6ccea4619b7183fbcc8c98bae1f9823d
+ENV LUA_DOWNLOAD "http://www.lua.org/ftp/lua-5.3.4.tar.gz"
+ENV LUA_MD5 53a9c68bcc0eda58bdc2095ad5cdfc63
+ENV PATH $PATH:/usr/libexec/s2i/
+
+LABEL io.k8s.description="Platform for building HAproxy" \
+      io.k8s.display-name="s2i HAproxy centos7 " \
+      io.openshift.expose-services="8080:http"\
+      io.openshift.tags="builder,s2i,haproxy,gateway,proxy,lua"
+
+RUN buildDeps='\
+        gcc \
+        make \
+        pcre-static \
+        pcre-devel \
+        openssl-devel \
+        readline-devel \
+        wget \
+    '\
+    && yum install -y ${buildDeps} \
+    && wget -O /tmp/lua.tar.gz ${LUA_DOWNLOAD} \
```

Comment:
It can't use Lua from CentOS ? Needs 5.3+ ?

Reply:
yes, haproxy requires lua 5.3+